pkg/lib/configreader/types.go

```diff
@@ -16,6 +16,77 @@ limitations under the License.
 
 package configreader
 
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	s "github.com/cortexlabs/cortex/pkg/lib/strings"
+)
+
 type TypePlaceholder struct {
 	Type string `json:"type"`
 }
+
+type PrimitiveType string
+type PrimitiveTypes []PrimitiveType
+
+var (
+	PrimTypeInt        PrimitiveType = "integer"
+	PrimTypeIntList    PrimitiveType = "integer list"
+	PrimTypeFloat      PrimitiveType = "float"
+	PrimTypeFloatList  PrimitiveType = "float list"
+	PrimTypeString     PrimitiveType = "string"
+	PrimTypeStringList PrimitiveType = "string list"
+	PrimTypeBool       PrimitiveType = "boolean"
+	PrimTypeBoolList   PrimitiveType = "boolean list"
+
+	PrimTypeMap     PrimitiveType = "map"
+	PrimTypeMapList PrimitiveType = "list of maps"
+	PrimTypeList    PrimitiveType = "list"
+
+	PrimTypeStringToStringMap PrimitiveType = "map of strings to strings"
+)
+
+func (ts PrimitiveTypes) StringList() []string {
+	strs := make([]string, len(ts))
+	for i, t := range ts {
+		strs[i] = string(t)
+	}
+	return strs
+}
+
+var (
+	typeStrRegex         = regexp.MustCompile(`"(INT|FLOAT|STRING|BOOL)(_COLUMN)?(\|(INT|FLOAT|STRING|BOOL)(_COLUMN)?)*"`)
+	singleDataTypeRegexp = regexp.MustCompile(`^\w*\w$`)
+)
+
+func EnvVar(envVarName string) string {
+	return fmt.Sprintf("environment variable \"%s\"", envVarName)
+}
+
+func DataTypeStrsOr(dataTypes []interface{}) string {
```

Comment:
Would it be easy to move the `DataType*` functions to `userconfig`? I'd rather keep Cortex related code out of `configreader` if possible